src/network/networkd-manager-bus.c

```diff
@@ -66,8 +67,11 @@ static int method_get_link_by_name(sd_bus_message *message, void *userdata, sd_b
                 return r;
 
         index = if_nametoindex(name);
-        if (index <= 0)
-                return sd_bus_error_setf(error, BUS_ERROR_NO_SUCH_LINK, "Link %s not known", name);
+        if (index <= 0) {
+                r = rtnl_resolve_link_alternative_name(&manager->rtnl, name, &index);
+                if (r < 0)
+                        return sd_bus_error_setf(error, BUS_ERROR_NO_SUCH_LINK, "Link %s not known", name);
+        }
```

Comment:
is this really necessary? I though that alternative names are already covered by if_nametoindex(), no?

@jpirko arent't they?

Reply:
It seems not yet.

Reply:
> is this really necessary? I though that alternative names are already covered by if_nametoindex(), no?
> 
> @jpirko arent't they?

They are not. The reason is if_nametoindex() is using legacy ioctl, however alternative names are only supported over rtnetlink.

Reply:
hmm, but why not support the alternative names in the ioctl too? i mean I am aware that SIOCGIFNAME only allows IFNAMSIZ names, but if_nametoindex() is really the official POSIX API for all this, so why not support it for at least those alternative names that fit into IFNAMSIZ?

i.e. all code that for example wants to bind an IPv6 link-local socket onto some iface is going to call if_nametoindex() to translate the iface name, but if the alternative names don't work then, then this severely limits their use...

Reply:
Doing that is very reasonable, but I think that at this point we don't have much choice, because even if the kernel is changed, it'll be a while.